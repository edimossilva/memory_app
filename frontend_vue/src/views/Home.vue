<template>
  <div class="home">
    <list-items class="m-4"></list-items>
    <section>
      <b-button
        data-cy="home__add_item_button"
        type="is-primary"
        @click="isComponentModalActive = true"
        >Add</b-button
      >
      <b-modal
        v-model="isComponentModalActive"
        has-modal-card
        trap-focus
        :destroy-on-hide="false"
        aria-role="dialog"
        aria-modal
      >
        <template #default="props">
          <add-item-modal @close="props.close"></add-item-modal>
        </template>
      </b-modal>
    </section>
  </div>
</template>

<script>
import ListItems from "@/components/item/ListItems.vue";
import AddItemModal from "@/modals/AddItemModal.vue";

export default {
  name: "Home",
  components: { ListItems, AddItemModal },
  data() {
    return {
      isComponentModalActive: false,
    };
  },
};
</script>
