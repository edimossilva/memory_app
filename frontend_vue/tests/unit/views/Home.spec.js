import { shallowMount } from "@vue/test-utils";
import Home from "@/views/Home.vue";
import Vue from "vue";
import VueRouter from "vue-router";

Vue.use(VueRouter);
describe("Home.vue", () => {
  it("Matchs Snapshot", () => {
    const wrapper = shallowMount(Home);
    expect(wrapper).toMatchSnapshot();
  });
  describe("methods", () => {
    describe("onClickAddButton()", () => {
      const show = jest.fn();
      const $modal = { show };
      const wrapper = shallowMount(Home, { mocks: { $modal } });

      beforeEach(() => {
        wrapper.vm.onClickAddButton();
      });

      it("should call push to dashboard", () => {
        expect(show).toHaveBeenNthCalledWith(1, "create-item-modal");
      });
    });
  });
});
