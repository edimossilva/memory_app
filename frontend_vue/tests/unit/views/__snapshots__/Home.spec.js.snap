// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue Matchs Snapshot 1`] = `
<div class="home">
  <list-items-stub class="m-4"></list-items-stub>
  <section>
    <b-button type="is-primary">Add</b-button>
    <b-modal has-modal-card="" trap-focus="" aria-role="dialog" aria-modal=""></b-modal>
  </section>
</div>
`;
